### PR TITLE
Optical Path Difference outputs

### PR DESCRIPTION
**Pull request by jmeyers314**
Adds ability to produce optical path difference (OPD) images as an extra output.  

Unit test compares directly with sample Zemax output.

### CODE REVIEW

**Comment by jmeyers314** [comment]
Added one more unrelated thing here: the ability to populate OpSimMeta directly from the config file.

**Comment by rmjarvis** [comment]
> Added one more unrelated thing here: the ability to populate OpSimMeta directly from the config file.

I'm not a huge fan of this change, tbh.  I feel like this is solidifying some technical debt that I'd rather try to get away from if possible.  I think, rather than have a class implicitly get something from the OpSimMeta dict and then let the user specify that dict with custom values, we should instead let them override the OpSimMeta value in that class's part of the config dict directly.

E.g.
Rather than:
```
input:
    opsim_meta_dict:
        meta:
            observationStartMJD: 9293

image:
    type: LSST_CCD
    [...]
```
We should instead enable:
```
image:
    type: LSST_CCD
    [...]
    observationStartMJD: 9293
```
Or possibly something like
```
image:
    type: LSST_CCD
    [...]
    header_values:  # Values for the output header to use in lieu of getting them from an opsim_meta_dict
        observationStartMJD: 9293
```
The last is probably my preference.  But either of these are better than making the user populate a fake OpSimMeta dict.  That's kind of abusing the meaning of what that is supposed to be.

**Review by rmjarvis — CHANGES_REQUESTED** [review]

File: imsim/opd.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        self.rotTelPos = kwargs.pop('rotTelPos', 0.0*galsim.radians)
+        self.nx = kwargs.pop('nx', 255)
+        self.projection = kwargs.pop('projection', 'postel')
+        self.reference = kwargs.pop('reference', 'chief')
+        self.telescope = GetInputObj(
+            'telescope',
+            config,
+            base,
+            'telescope'
+        ).fiducial
```
[the remark this answers is not in the document]
Comment (jmeyers314) [reply imsim/opd.py:86]: Hmm..  I was placing it here to "fail fast" if there's a config error.  E.g., if the telescope doesn't have a `sphereRadius` attribute and one isn't specified in the config either.  I could reload the telescope it in `finalize` I suppose.

[the remark this answers is not in the document]
Comment (jmeyers314) [reply imsim/opd.py:86]: Done.  Telescope is reloaded during finalize.

File: imsim/opd.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        # Handle defaults for optional kwargs
+        self.rotTelPos = kwargs.pop('rotTelPos', 0.0*galsim.radians)
+        self.nx = kwargs.pop('nx', 255)
+        self.projection = kwargs.pop('projection', 'postel')
+        self.reference = kwargs.pop('reference', 'chief')
+        telescope = GetInputObj(
+            'telescope',
+            config,
+            base,
+            'telescope'
```
Comment (rmjarvis) [inline imsim/opd.py:85]: Oh, one other extremely minor thing I just noticed.  The last thing here should be 'opd', not 'telescope'.  It's only used for error messages, saying which thing needed the telescope input item if it fails to load.

**Review by rmjarvis — APPROVED** [review]

**Pull request merged**